packages/models/src/lib/table.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from 'zod';
+
+export const tableHeadingSchema = z.object(
+  {
+    key: z.string(),
+    label: z.string().optional(),
+  },
+  { description: 'Table heading' },
+);
+export type TableHeading = z.infer<typeof tableHeadingSchema>;
+
+export const tableAlignmentSchema = z.enum(['l', 'c', 'r'], {
+  description: 'Cell alignment',
+});
+export type TableAlignment = z.infer<typeof tableAlignmentSchema>;
+export const primitiveValueSchema = z.union([z.string(), z.number()]);
+export type PrimitiveValue = z.infer<typeof primitiveValueSchema>;
+
+export const tableSchema = (description = 'Table information') =>
+  z.object(
+    {
+      headings: z.array(tableHeadingSchema).optional(),
+      alignment: z.array(tableAlignmentSchema).optional(),
+      rows: z.array(
+        z.union([
+          z.array(primitiveValueSchema),
+          z.record(primitiveValueSchema),
+        ]),
+      ),
+    },
+    { description },
+  );
```

Comment:
Had a thought. The headings and alignments are both related to columns and should both have the same length. So why won't we merge them? :thinking: 

- headings with alignment can be specified with `columns: [{ key: 'foo', label: 'Foo', align: 'l' }]
- only alignment can be specified with `columns: ['l', 'c', 'c;]`

```suggestion
export const tableSchema = (description = 'Table information') =>
  z.object(
    {
      rows: z.array(
        z.union([
          z.array(primitiveValueSchema),
          z.record(primitiveValueSchema),
        ]),
      ),
      columns: z
        .union([
          z.array(
            z.object({
              key: z.string(),
              label: z.string().optional(),
              align: tableAlignSchema.optional(),
            }),
          ),
          z.array(tableAlignSchema),
        ])
        .optional(),
    },
    { description },
  );
```

Reply:
opened issue for this and the generic type for row:
https://github.com/code-pushup/cli/issues/637

Move it into another PR.

Reply:
OK, added a comment in the issue: https://github.com/code-pushup/cli/issues/637#issuecomment-2071704785